central/policycategory/search/searcher_impl.go

```diff
@@ -0,0 +1,107 @@
+package search
+
+import (
+	"context"
+
+	"github.com/stackrox/rox/central/policycategory/index"
+	categoryMapping "github.com/stackrox/rox/central/policycategory/index/mappings"
+	"github.com/stackrox/rox/central/policycategory/store"
+	"github.com/stackrox/rox/central/role/resources"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/search"
+	"github.com/stackrox/rox/pkg/search/blevesearch"
+	"github.com/stackrox/rox/pkg/search/paginated"
+	"github.com/stackrox/rox/pkg/search/sortfields"
+)
+
+var (
+	defaultSortOption = &v1.QuerySortOption{
+		Field: search.PolicyCategoryName.String(),
+	}
+
+	policySAC = sac.ForResource(resources.Policy)
+)
+
+// searcherImpl provides an intermediary implementation layer for AlertStorage.
+type searcherImpl struct {
+	storage  store.Store
+	indexer  index.Indexer
+	searcher search.Searcher
+}
+
+func (s *searcherImpl) Search(ctx context.Context, q *v1.Query) ([]search.Result, error) {
+	if ok, err := policySAC.ReadAllowed(ctx); err != nil || !ok {
+		return nil, err
+	}
+
+	return s.searcher.Search(ctx, q)
+}
+
+func (s *searcherImpl) Count(ctx context.Context, q *v1.Query) (int, error) {
+	if ok, err := policySAC.ReadAllowed(ctx); err != nil || !ok {
+		return 0, err
+	}
+
+	return s.searcher.Count(ctx, q)
+}
+
+func (s searcherImpl) SearchRawCategories(ctx context.Context, q *v1.Query) ([]*storage.PolicyCategory, error) {
+	categories, _, err := s.searchCategories(ctx, q)
+	return categories, err
+}
+
+func (s searcherImpl) SearchCategories(ctx context.Context, q *v1.Query) ([]*v1.SearchResult, error) {
+	categories, results, err := s.searchCategories(ctx, q)
+	if err != nil {
+		return nil, err
+	}
+	protoResults := make([]*v1.SearchResult, 0, len(categories))
+	for i, c := range categories {
+		protoResults = append(protoResults, convertCategory(c, results[i]))
+	}
+	return protoResults, nil
+}
+
+func (s *searcherImpl) searchCategories(ctx context.Context, q *v1.Query) ([]*storage.PolicyCategory, []search.Result, error) {
+	results, err := s.Search(ctx, q)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var categories []*storage.PolicyCategory
+	var newResults []search.Result
+	for _, result := range results {
+		category, exists, err := s.storage.Get(ctx, result.ID)
```

Comment:
This seems like it would be more efficient to get many then filter out by the missing indices